sklearn/tree/tests/test_tree.py

```diff
@@ -1325,3 +1335,44 @@ def test_presort_sparse():
 
     for est, sparse_matrix in product(ests, sparse_matrices):
         yield check_presort_sparse, est, sparse_matrix(X), y
+
+
+def test_decision_path_hardcoded():
+    X = iris.data
+    y = iris.target
+    est = DecisionTreeClassifier(random_state=0, max_depth=1).fit(X, y)
+    node_indicator = est.decision_path(X[:2]).toarray()
+    assert_array_equal(node_indicator, [[1, 1, 0], [1, 0, 1]])
```

Comment:
I think I am misunderstanding these paths. How can the first point go to node 1, then node 1 again, then node 0? My understanding was that the decision path was an array of node IDs, ending in a leaf node.